### PR TITLE
fix: silent NaN to zero fill

### PR DESCRIPTION
**Pull request by bagustris**
* fix: elevate NaN handling from debug to warning with configurable strategy

- Add Util.handle_nan() method supporting zero/mean/median/drop strategies
- Add [FEATS] nan_strategy INI config option (default: zero)
- Replace all debug-level NaN messages with warnings including count and %
- Update models: model.py, model_xgb.py, model_mlp.py, model_mlp_regression.py, model_adm.py
- Update feature extractors: feats_import.py, feats_praat.py, feats_analyser.py
- Update utility functions: stats.py, feats_praat_core.py
- Add comprehensive tests for handle_nan in tests/utils/test_util.py

* fix: add zero fallback for mean/median strategy when all values are NaN

Address code review feedback:
- Chain .fillna(0) after mean/median fillna to handle all-NaN columns
- Add test for all-NaN column edge case

* style: fix isort and black formatting issues

Run isort --profile black and black on all modified files to pass CI checks.

* fix: move inline imports to module level to fix SonarQube reliability

* fix: address code review - add NaN fallback comments, add test cleanup fixture

* fix: move inline imports to module level to resolve SonarQube C reliability

* fix: move inline imports to module level in TestSetupLogging to resolve SonarQube C reliability

* fix: address NaN review and PR 52 CI failures

* refactor: centralize model NaN handling

---------

### CODE REVIEW

File: nkululeko/utils/util.py [hunk 1]
```diff
@@ -265,6 +280,61 @@ def debug(self, message):
...
+            strategy = "zero"
+
+        self.warn(
+            f"{context}: replacing {nan_count} NaN values"
+            f" ({nan_pct:.1f}% of data) with strategy '{strategy}'"
+        )
+
+        if strategy == "mean":
+            # Second fillna(0) handles columns where all values are NaN (mean is NaN)
+            numeric_means = df.mean(numeric_only=True)
```
Comment (felixbur) [inline nkululeko/utils/util.py:326]: shouldn't that be 

mean(axis=1)

so that Nan are filled with column mean and not sample mean?

Comment (bagustris) [reply nkululeko/utils/util.py:326]: Will take a look this

Comment (bagustris) [reply nkululeko/utils/util.py:326]: Thanks for the review, Felix! I think the current `axis=0` (the pandas default) is actually what we want here. `df.mean(numeric_only=True)` computes the mean along `axis=0`, which collapses across rows and produces one mean value per **column** (i.e., per feature). Then `df.fillna(numeric_means)` fills each NaN cell with its column's mean — which is the standard approach for feature imputation.

Using `axis=1` would compute the mean along the column axis, producing one mean value per **row** (i.e., per sample). That would mix values across features with different scales and units, which is not desirable for imputation.

That said, let me know if you had a different use case in mind and I'm happy to revisit!

Comment (felixbur) [reply nkululeko/utils/util.py:326]: you're right, sorry for the confuion
